### ISSUE TITLE
anaconda-project does not encrypt variables 

### ISSUE THREAD

**Issue by DavidMertz** [Anaconda-Platform/anaconda-project#141]
Filed earlier in less good: https://github.com/ContinuumIO/anaconda-issues/issues/7459
---
More with certification/course for `anaconda-project`.  Notwithstanding the documentation stating:

> Any variable ending in _PASSWORD, _SECRET, or _SECRET_KEY is automatically encrypted.

I find the following:

```bash
684-tmp % anaconda-project init --directory secrets --yes
Project configuration is in /Users/dmertz/tmp/secrets/anaconda-project.yml
685-tmp % cd secrets/
686-secrets % anaconda-project add-variable --default=user USERNAME
687-secrets % anaconda-project add-variable --default=pw USER_PASSWORD
688-secrets % grep variables: anaconda-project.yml
variables: {USERNAME: user, USER_PASSWORD: pw}
```

Doesn’t look very encrypted to me.  I've also tried it with `_SECRET` suffix, same story.

**Comment by fpliger** [comment]
@DavidMertz was looking into it.. I think it's a different issue. You are running (as described in the docs):

```
anaconda-project add-variable --default=pw USER_PASSWORD
```

This does not work for me either, while:

```
anaconda-project add-variable USER_PASSWORD --default=pw
```

does. Would you give it a try? :)

**Comment by DavidMertz** [comment]
Putting the switch at end:

```bash
747-tmp % anaconda-project init --directory secrets --yes
Project configuration is in /Users/dmertz/tmp/secrets/anaconda-project.yml
748-tmp % cd secrets/
749-secrets % anaconda-project add-variable USERNAME --default=user
750-secrets % anaconda-project add-variable USER_PASSWORD --default=pw
751-secrets % grep variables: anaconda-project.yml
variables: {USERNAME: !!null '', --default=user: !!null '', USER_PASSWORD: !!null '',
752-secrets % ll
total 16
drwxr-xr-x   4 dmertz  staff   136B Nov 28 11:43 .
drwxr-xr-x  54 dmertz  staff   1.8K Nov 28 11:42 ..
-rw-r--r--   1 dmertz  staff   231B Nov 28 11:42 .projectignore
-rw-r--r--   1 dmertz  staff   2.0K Nov 28 11:43 anaconda-project.yml
```

That's arguably **better**, but still not right.  `USERNAME` should be stored with a plaintext value.  Instead it is stored as `!!null` and also the `--default` switch is stored as if it is another variable.

**Comment by fpliger** [comment]
> That's arguably better, but still not right.

Yeah, as I said, think _it's a different issue_ but  still an issue. ;)

**Comment by DavidMertz** [comment]
I don't think putting the switch at the end actually causes the password to be stored in an encrypted way for `*_PASSWORD`.  I think it just treats it as a "user must specify at runtime" variable.

**Comment by fpliger** [comment]
Yeah... will have to look better at it. Might be the case that we have more than issue here. Will check with @goanpeca who has more bandwidth to look into it

**Comment by goanpeca** [comment]
>Might be the case that we have more than issue here. Will check with @goanpeca who has more bandwidth to look into it

Really :-p ?

**Comment by fpliger** [comment]
@goanpeca  Yeap :)

**Comment by goanpeca** [comment]
Hmmm strange, I haven't been formally notified of this :trollface:

**Comment by fpliger** [comment]
well.. this _is_ a formal notification. :)

**Comment by AlbertDeFusco** [comment]
I'm using v0.8.2 and I found a two work-arounds.

1. `anaconda-project add-variable MY_PASSWORD`
1. `anaconda-project set-variable MY_PASSWORD=pass`

or

1. `anaconda-project add-variable MY_PASSWORD`
1. `anaconda-project run` and it will ask for a `MY_PASSWORD`. 

When I have `keyring` installed either of the step 2 lines adds the password to my OSX Keychain. If I don't have `keyring` installed then `anaconda-project run` will always ask for `MY_PASSWORD`.

**Comment by mcg1969** [comment]
Thanks for opening this issue, and apologies for the long delay in responding.

Anaconda Project has not been actively developed in some time and has largely been superseded by newer alternatives:

* [pixi](https://pixi.sh/) and [uv](https://docs.astral.sh/uv/) represent the first truly successful successors.
* [conda-workspaces](https://github.com/conda-incubator/conda-workspaces) is a promising pixi-compatible alternative for users who prefer to stay closer to the conda ecosystem.

Going forward, work on this repository is focused on critical bug fixes and on tooling to help users migrate existing Anaconda Project projects to pixi (and, by extension, conda-workspaces). See the [Pixi support](https://anaconda-project.readthedocs.io/en/latest/pixi-support.html) page in our documentation for details on the `anaconda-project export-pixi` command, which can convert an existing `anaconda-project.yml` into a `pixi.toml` for you.

The `anaconda-project` package itself will remain available, so existing projects relying on its stable feature set can continue to be supported. However, we do not anticipate further enhancement work, and for that reason we are closing out the bulk of our open issue backlog — including this one.

If this issue describes a critical regression in the stable feature set (as opposed to an enhancement or an edge case in a feature we no longer plan to expand) and you have a minimal reproducer against a current release, please feel free to reopen and we will take another look.

Thank you for your interest in Anaconda Project.
